tasks/release.py

```diff
@@ -533,17 +559,131 @@ def _get_highest_version_from_release_json(release_json, highest_major, version_
+# This is here for compatibility and simplicity reasons, as in most repos
+# we don't create both 6 and 7 tags for a combined Agent 6 & 7 release.
+# The order matters, eg. when fetching matching tags for an Agent 6 entry,
+# tags starting with 6 will be preferred to tags starting with 7.
+COMPATIBLE_MAJOR_VERSIONS = {6: ["6", "7"], 7: ["7"]}
+
+# Message templates for the below functions
+# Defined here either because they're long and would make the code less legible,
+# or because they're used multiple times.
+DIFFERENT_TAGS_TEMPLATE = (
+    "The latest version of {} ({}) does not match the latest version found in the release.json file ({})."
+)
+TAG_NOT_FOUND_TEMPLATE = "Couldn't find a(n) {} version compatible with the new Agent version entry {}"
+RC_TAG_QUESTION_TEMPLATE = "The {} tag found is an RC tag: {}. Are you sure you want to use it?"
+TAG_FOUND_TEMPLATE = "The {} tag is {}"
+
+
+##
+## Repository version fetch functions
+## The following functions aim at returning the correct version to use for a given
+## repository, after compatibility & user confirmations
+## The version object returned by such functions should be ready to be used to fill
+## the release.json entry.
+##
+
+
+def _fetch_dependency_repo_version(
+    repo_name, new_agent_version, allowed_major_versions, compatible_version_re, github_token, check_for_rc
+):
+    """
+    Fetches the latest tag on a given repository whose version scheme matches the one used for the Agent,
+    with the following constraints:
+    - the tag must have a major version that's in allowed_major_versions
+    - the tag must match compatible_version_re (the main usage is to restrict the compatible tags to the
+      ones with the same minor version as the Agent)?
+    
+    If check_for_rc is true, a warning will be emitted if the latest version that satisfies
+    the constraints is an RC. User confirmation is then needed to check that this is desired.
+    """
+
+    version = _get_highest_repo_version(github_token, repo_name, "", allowed_major_versions, compatible_version_re)
+    if not version:
+        raise Exit(TAG_NOT_FOUND_TEMPLATE.format(repo_name, _stringify_version(new_agent_version)), 1)
+    if check_for_rc and version["rc"] is not None:
+        if not yes_no_question(
+            RC_TAG_QUESTION_TEMPLATE.format(repo_name, _stringify_version(version)), "orange", False,
+        ):
+            raise Exit("Aborting release.json update.", 1)
+
+    print(TAG_FOUND_TEMPLATE.format(repo_name, _stringify_version(version)))
+    return version
+
+
+def _confirm_independent_dependency_repo_version(repo, latest_version, highest_release_json_version):
+    """
+    Checks if the two versions of a repository we found (from release.json and from the available repo tags)
+    are different. If they are, asks the user for confirmation before updating the version.
+    """
+
+    confirmed_version = highest_release_json_version
+    if latest_version != highest_release_json_version:
```

Comment:
I feel this would be clearer to me if we did this the other way around:

```
if latest_version  == highest_release_json_version:
    return highest_release_json_version
```

and then do the rest outside of the `if`. That way we have less indentation